core/sodasql/scan/sampler.py

```diff
@@ -150,66 +150,32 @@ def __serialize_file_upload_value(self, value):
             return value
         return str(value)
 
-    def __get_sample_columns(self, cursor):
+    def __get_sample_columns(self, description):
         return [
             {'name': d[0],
              'type': self.scan.warehouse.dialect.get_type_name(d)}
-            for d in cursor.description
+            for d in description
         ]
 
-    def save_sample_to_local_file(self, sql, temp_file):
-        cursor = self.scan.warehouse.connection.cursor()
-        try:
-            logger.debug(f'Executing SQL query: \n{sql}')
-            start = datetime.now()
-            cursor.execute(sql)
-            sample_columns = self.__get_sample_columns(cursor)
-
-            stored_rows = 0
-            row = cursor.fetchone()
-            while row is not None:
-                sample_values = []
-                for i in range(0, len(row)):
-                    sample_values.append(self.__serialize_file_upload_value(row[i]))
-                temp_file.write(bytearray(json.dumps(sample_values), 'utf-8'))
-                temp_file.write(b'\n')
-                stored_rows += 1
-                row = cursor.fetchone()
-
-            delta = datetime.now() - start
-            logger.debug(f'SQL took {str(delta)}')
-
-        finally:
-            cursor.close()
-
-        return stored_rows, sample_columns
+    def save_sample_to_local_file(self, sql, temp_file, *, limit = None):
+        logger.debug('Executing SQL query: \n%s', sql)
```

Comment:
I think we should add this to the methods on the warehouse instead of doing this logging here.